### ISSUE TITLE
Installing pyauditor 0.3.0 from source results in broken install

### ISSUE THREAD

**Issue by QuantumDancer** [ALU-Schumacher/AUDITOR#565]
When installing pyauditor v0.3.0 from source the resulting install is missing all the python classes:

```
$ python -m venv .venv
$ source .venv/bin/activate
$ pip install python-auditor==0.3.0 --no-binary python-auditor
Collecting python-auditor==0.3.0
  Using cached python_auditor-0.3.0.tar.gz (71 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: python-auditor
  Building wheel for python-auditor (pyproject.toml) ... done
  Created wheel for python-auditor: filename=python_auditor-0.3.0-cp311-cp311-linux_x86_64.whl size=3262657 sha256=344c63cf96b427611e391e43bce1eb173fa831c338d9e64d1ad8548bb505a4e8
  Stored in directory: /home/benjamin/.cache/pip/wheels/34/6e/b0/23ae13789c39cc180a4024b27aaa63108f5bba5f6f21e6cb03
Successfully built python-auditor
Installing collected packages: python-auditor
Successfully installed python-auditor-0.3.0
$ python                                                      
Python 3.11.5 (main, Sep  2 2023, 14:16:33) [GCC 13.2.1 20230801] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyauditor
>>> dir(pyauditor)
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```

For v0.2.0, things look fine:

```
$ python -m venv .venv
$ source .venv/bin/activate
$ pip install python-auditor==0.2.0 --no-binary python-auditor
Collecting python-auditor==0.2.0
  Downloading python_auditor-0.2.0.tar.gz (70 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 70.2/70.2 kB 2.4 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: python-auditor
  Building wheel for python-auditor (pyproject.toml) ... done
  Created wheel for python-auditor: filename=python_auditor-0.2.0-cp311-cp311-linux_x86_64.whl size=4958646 sha256=9e7ca4c68c7b89d261dc27455db61dd32842b6fcd283a41135c955a0c2e64d64
  Stored in directory: /home/benjamin/.cache/pip/wheels/46/56/19/0e472647f48ec64229f3f1d88a63ad2678408e2e6981fc619f
Successfully built python-auditor
Installing collected packages: python-auditor
Successfully installed python-auditor-0.2.0
$ python                                                      
Python 3.11.5 (main, Sep  2 2023, 14:16:33) [GCC 13.2.1 20230801] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyauditor
>>> dir(pyauditor)
['AuditorClient', 'AuditorClientBlocking', 'AuditorClientBuilder', 'Component', 'Meta', 'Record', 'Score', '__all__', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'pyauditor']
```

This issue was first discovered in https://github.com/MatterMiners/tardis/issues/320 by @giffels

**Comment by QuantumDancer** [comment]
So the pre-built wheels and the wheel resulting from building from source seem to differ:

pre-built wheel:
```
python_auditor-0.3.0
├── pyauditor
│   ├── __init__.py
│   └── pyauditor.cpython-311-x86_64-linux-gnu.so
└── python_auditor-0.3.0.dist-info
    ├── METADATA
    ├── RECORD
    └── WHEEL
```

from source:
```
python_auditor-0.3.0
├── pyauditor
│   ├── Cargo.toml
│   ├── docs
│   │   ├── api.rst
│   │   ├── changelog.rst
│   │   ├── conf.py
│   │   ├── examples.rst
│   │   ├── index.rst
│   │   ├── make.bat
│   │   └── Makefile
│   ├── pyauditor.cpython-311-x86_64-linux-gnu.so
│   ├── README.md
│   ├── scripts
│   │   ├── test_add_update_blocking.py
│   │   ├── test_add_update.py
│   │   ├── test_components_blocking.py
│   │   ├── test_components.py
│   │   ├── test_eq.py
│   │   ├── test_get_since_blocking.py
│   │   ├── test_get_since.py
│   │   ├── test_meta_blocking.py
│   │   └── test_meta.py
│   └── src
│       ├── blocking_client.rs
│       ├── builder.rs
│       ├── client.rs
│       ├── domain
│       │   ├── component.rs
│       │   ├── meta.rs
│       │   ├── mod.rs
│       │   ├── record.rs
│       │   └── score.rs
│       └── lib.rs
└── python_auditor-0.3.0.dist-info
    ├── METADATA
    ├── RECORD
    └── WHEEL
```

The `__init__.py` file seems to be missing in the wheel that is built from source. This also explains, why nothing is loaded when importing pyauditor.

**Comment by QuantumDancer** [comment]
Ok, I tried to build the pyauditor sdist for different auditor and [maturin versions](https://github.com/PyO3/maturin/releases):

| auditor | maturin | working |
|---------|---------|--------------------|
| 0.2.0   | 1.3.2   | :x:                |
| 0.3.0   | 1.2.3   | :white_check_mark: |
| 0.3.0   | 1.3.0   | :x:                |
| 0.3.0   | 1.3.2   | :x:                |

So this is clearly an issue with maturin. I propose that we fix the maturin version to 1.2.3 and release a bugfix version of auditor. Then we can further investigate why this is not working with the latest maturin version and eventually file a bug report.
